tests/Teams/ActionBasedMessagingExtension-FetchTask/Bots/ActionBasedMessagingExtensionFetchTaskBot.cs

```diff
@@ -58,11 +51,31 @@ protected override async Task<MessagingExtensionActionResponse> OnTeamsMessaging
 
         protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionBotMessagePreviewSendAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action, CancellationToken cancellationToken)
         {
-            var reply = MessageFactory.Text("OnTeamsMessagingExtensionBotMessagePreviewSendAsync MessagingExtensionAction: " + JsonConvert.SerializeObject(action));
-            await turnContext.SendActivityAsync(reply, cancellationToken);
-
             var submitData = action.ToSubmitExampleData();
             var adaptiveCard = submitData.ToAdaptiveCard();
+            var responseActivity = Activity.CreateMessageActivity();
+            Attachment attachment = new Attachment()
+            {
+                ContentType = AdaptiveCard.ContentType,
+                Content = adaptiveCard,
+            };
+            responseActivity.Attachments.Add(attachment);
+            try
+            {
+                // Send to channel where messaging extension invoked.
+                var channelData = turnContext.Activity.GetChannelData<TeamsChannelData>();
+                await turnContext.TeamsCreateConversationAsync(channelData.Channel.Id, responseActivity);
```

Comment:
This extension method can replace the retrieval of TeamsChannelData:

https://github.com/microsoft/botbuilder-dotnet/blob/eef0b44af623e0cad2ec52dc7b041498812e3cef/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityExtensions.cs#L11

Reply:
Nice, thanks!